flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraph.java

```diff
@@ -199,6 +199,7 @@ public StateBackend getStateBackend() {
 
     public void setChangelogStateBackendEnabled(TernaryBoolean changelogStateBackendEnabled) {
         this.changelogStateBackendEnabled = changelogStateBackendEnabled;
+        this.executionConfig.setChangelogStateBackendEnabled(changelogStateBackendEnabled);
```

Comment:
I think `changelogStateBackendEnabled` field is unnecessary now and its read can be replaced with `executionConfig.isChangelogStateBackendEnabled()`.